### DISCUSSION TITLE
GitHub Copilot Not working

### DISCUSSION THREAD

**Discussion by Ygame20** [community/community#11080] [Copilot Conversations]
My GitHub Copilot has already activate but I can't use it after I restart my Visual Studio Code.
![image](https://user-images.githubusercontent.com/94981201/152666957-e404fe9a-0551-4e33-a152-af51ab347469.png)

**Comment by davecheney** [comment]
What do you see when you hover over the red copilot icon?

**Comment by shlomiandadi** [comment]
Restart the plugin

**Comment by collinsnji** [comment]
I'm having the same issue as OP. When I switched to the `Log(Extension Host)` logs, this is what I see
```
[2022-02-06 16:44:21.763] [exthost] [error] [GitHub.copilot] provider FAILED
[2022-02-06 16:44:21.763] [exthost] [error] TypeError: Converting circular structure to JSON
    --> starting at object with constructor 't.Context'
    |     property 'instances' -> object with constructor 'Array'
    |     index 16 -> object with constructor 'Array'
    |     index 1 -> object with constructor 't.CopilotStatusBar'
    --- property 'ctx' closes the circle
    at JSON.stringify (<anonymous>)
    at /Users/<username>/.vscode/extensions/github.copilot-1.7.4812/dist/extension.js:8:568108
    at t.extractRepoInfoInBackground (/Users/<username>/.vscode/extensions/github.copilot-1.7.4812/dist/extension.js:8:568012)
    at /Users/<username>/.vscode/extensions/github.copilot-1.7.4812/dist/extension.js:8:537782
    at t.getGhostText (/Users/<username>/.vscode/extensions/github.copilot-1.7.4812/dist/extension.js:8:538282)
    at f (/Users/<username>/.vscode/extensions/github.copilot-1.7.4812/dist/extension.js:8:493071)
    at q.provideInlineCompletions (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:93:106825)
```
Here's what I have tried (and failed) to fix the problem
- Restarted VS Code multiple times
- Disabled and reenabled the extension
- Disable for specific languages only (Python and C)
- Disabled/reenable in workspace and globally

VSCode Version:
```
1.64.0
5554b12acf27056905806867f251c859323ff7e9
x64
```

OS: 
`MacBook Pro (Fall 2017) with macOS Monterey v12.1`

**Reply by kashishgrover** [reply]
```
[2022-02-07 12:07:00.047] [exthost] [error] TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'Timeout'
    |     property '_idlePrev' -> object with constructor 'TimersList'
    --- property '_idleNext' closes the circle
    at JSON.stringify (<anonymous>)
    at /Users/<username>/.vscode/extensions/github.copilot-1.7.4812/dist/extension.js:8:568108
    at t.extractRepoInfoInBackground (/Users/<username>/.vscode/extensions/github.copilot-1.7.4812/dist/extension.js:8:568012)
    at /Users/<username>/.vscode/extensions/github.copilot-1.7.4812/dist/extension.js:8:537782
    at t.getGhostText (/Users/<username>/.vscode/extensions/github.copilot-1.7.4812/dist/extension.js:8:538282)
    at f (/Users/<username>/.vscode/extensions/github.copilot-1.7.4812/dist/extension.js:8:493048)
    at K.provideInlineCompletions (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:95:103883)
```

Seeing similar errors. A lot of them.

```
Version: 1.63.2 (Universal)
Commit: 899d46d82c4c95423fb7e10e68eba52050e30ba3
Date: 2021-12-15T09:37:28.172Z (1 mo ago)
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 20.6.0
```

**Comment by kashishgrover** [comment]
<img width="339" alt="Screenshot 2022-02-07 at 12 26 33 PM" src="https://user-images.githubusercontent.com/13067699/152739578-890e7a7a-7f23-470c-865a-127eebd7d3cd.png">

<img width="756" alt="Screenshot 2022-02-07 at 12 26 27 PM" src="https://user-images.githubusercontent.com/13067699/152739591-4eda5260-9fdc-41a2-933b-f87432f8263a.png">

Current version is broken it seems. I installed an older version and it seems to be working fine now

**Reply by collinsnji** [reply]
I installed an older version too and it seems to have resolved the issue. This is the older version I installed:
```
Name: GitHub Copilot
Id: GitHub.copilot
Description: Your AI pair programmer
Version: 1.7.4406
Publisher: GitHub
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=GitHub.copilot
```

**Comment by Stay1444** [comment]
Rider has the same issue. And IntelliJ idea too. Last update broke it, now i have to reinstall the plugin everytime i open my IDE

**Comment by santiago-mooser** [comment]
The latest update also made the extension stop working for me, but @kashishgrover's answer worked for me. I've downgraded to 1.7.4812.

@Stay1444 You can disable automatic updates in VSCode so you get notified of them but require manual installs. 


Here are the details of the setup that **does not** work.

Copilot:
```
Name: GitHub Copilot
Id: GitHub.copilot
Description: Your AI pair programmer
Version: 1.7.5209
Publisher: GitHub
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=GitHub.copilot
```

VS Code:
```
Version: 1.65.0
Commit: b5205cc8eb4fbaa726835538cd82372cc0222d43
Date: 2022-03-02T11:12:36.248Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Linux x64 5.15.25-2-lts
```

**Comment by kashishgrover** [comment]
I'm not able to use it now because the older version doesn't work anymore. I set up vscode and copilot in one of my other devices and it worked. But it fails to work on my primary work machines. Very disappointing :/

**Comment by kashishgrover** [comment]
Aah so it was bothering me and I unminified the extension.js file for the plugin. Looks like there's a `JSON.stringify()` crashing over a cyclic reference. Commenting out the code in that function `extractRepoInfoInBackground` makes the plugin run again.

Refer the following piece of unminified code:

```
  }, t.extractRepoInfoInBackground = function(e, t) {
      return;
      // if (!t) return;
      // const r = (0, n.dirname)(t);
      // return a(e, r)
  };
  const a = function(e, t) {
        const r = new o.LRUCache(1e4),
            i = new Set;
        return (...e) => {
            const t = JSON.stringify(e),  <------ CRASHING
                o = r.get(t);
  ```
  
I unminified the file using https://www.unminify2.com/ by copying the contents of the `~/.vscode/extensions/github.copilot-1.30.6165/dist/extensions.js` file. Note that the 30k lines of unminified result might first break the chrome tab and then break vscode for you

**Reply by bmuskalla** [reply]
Thanks for the report and the investigation @kashishgrover - we're investigating the root issue and hope to get a release out the door as soon as we have a proper fix for this

**Reply by JohnMeyerhoff** [reply]
@bmuskalla Hi, please do keep in mind what I added yesterday. Removing the specific extension yielded good results and also was reproducible. Adding the extension back in, the cyclic error reappeared. There is most likely nothing directly wrong with github copilot itself.

**Reply by kashishgrover** [reply]
> @bmuskalla Hi, please do keep in mind what I added yesterday. Removing the specific extension yielded good results and also was reproducible. Adding the extension back in, the cyclic error reappeared. There is most likely nothing directly wrong with github copilot itself.

I was able to reproduce this each time with only GitHub copilot enabled as the extension. It may be the case with you, but I was able to reproduce this each time with and without other extensions enabled

**Comment by JohnMeyerhoff** [comment]
For me it took really long to figure out but it was this extension:
``iagolaguna.vscodefy``
which i had install. My advice for your troubleshooting:
- disable all extensions
- reload
- enable copilot. 

Get copilot working as the only extension, if it works, slowly enable extensions and see when the debug console gives you the warning.
